langchain/src/llms/azure_ml.ts

```diff
@@ -0,0 +1,200 @@
+export interface ContentFormatter {
+    /**
+   * Formats the request payload for the AzureML endpoint. It takes a
+   * prompt and a dictionary of model arguments as input and returns a
+   * string representing the formatted request payload.
+   * @param prompt The prompt for the AzureML model.
+   * @param modelArgs A dictionary of model arguments.
+   * @returns A string representing the formatted request payload.
+   */
+  formatRequestPayload:(prompt:string, modelArgs:Record<string, unknown>) => string;
+  /**
+   * Formats the response payload from the AzureML endpoint. It takes a
+   * response payload as input and returns a string representing the
+   * formatted response.
+   * @param responsePayload The response payload from the AzureML endpoint.
+   * @returns A string representing the formatted response.
+   */
+  formatResponsePayload: (output: string) => string;
+}
+
+export class GPT2ContentFormatter implements ContentFormatter {
+  formatRequestPayload(prompt: string, modelArgs: Record<string, unknown>): string {
+    return JSON.stringify({
+      inputs: {
+        input_string: [prompt],
+      },
+      parameters: modelArgs,
+    });
+  };
+  formatResponsePayload(output: string): string {
+    return JSON.parse(output)[0]["0"]
+  };
+}
+
+export class HFContentFormatter implements ContentFormatter {
+  formatRequestPayload(prompt: string, modelArgs: Record<string, unknown>): string {
+    return JSON.stringify({
+      inputs: [prompt],
+      parameters: modelArgs,
+    });
+  };
+  formatResponsePayload(output: string): string {
+    return JSON.parse(output)[0]["generated_text"]
+  };
+}
+
+export class DollyContentFormatter implements ContentFormatter {
+  formatRequestPayload(prompt: string, modelArgs: Record<string, unknown>): string {
+    return JSON.stringify({
+      input_data: {
+        input_string: [prompt],
+      },
+      parameters: modelArgs,
+    });
+  };
+  formatResponsePayload(output: string): string {
+    return JSON.parse(output)[0]
+  };
+}
+
+export class LlamaContentFormatter implements ContentFormatter {
+  formatRequestPayload(prompt: string, modelArgs: Record<string, unknown>): string {
+    return JSON.stringify({
+      input_data: {
+        input_string: [prompt],
+      },
+      parameters: modelArgs,
+    });
+  };
+  formatResponsePayload(output: string): string {
+    return JSON.parse(output)[0]["0"]
+  };
+}
+
+export interface AzureMLParams extends BaseLLMParams {
+  endpointUrl?: string;
+  endpointApiKey?: string;
+  deploymentName?: string;
+  modelArgs?: Record<string, unknown>;
+  contentFormatter?: ContentFormatter;
+}
+
+/**
+ * Class that represents an AzureML model. It extends the LLM base class
+ * and provides methods for calling the AzureML endpoint and formatting
+ * the request and response payloads.
+ */
+export class AzureMLModel extends LLM implements AzureMLParams {
+  _llmType() {
+    return "azure_ml";
+  }
+  static lc_name() {
+    return "AzureMLModel";
```

Comment:
See above, should also change this to match Python class name for better serialization